pkg/cli/common/prerun/prerun.go

```diff
@@ -48,5 +50,27 @@ func InitClient(p common.Params, cmd *cobra.Command) (*client.RESTClient, error)
 	if err != nil {
 		return nil, err
 	}
-	return client.NewRESTClient(c.Get())
+
+	// Check if the configuration is properly set up
+	clientConfig := c.Get()
+	if clientConfig == nil || clientConfig.URL == nil {
+		return nil, fmt.Errorf("Configuration is not set up. Please run 'config set' command first to configure the CLI")
+	}
+
+	// Check if essential configuration values are missing
+	extensionObj := c.GetObject()
+	if extensionObj == nil {
+		return nil, fmt.Errorf("Configuration is not set up. Please run 'config set' command first to configure the CLI")
+	}
+
+	extension, ok := extensionObj.(*config.Extension)
+	if !ok {
+		return nil, fmt.Errorf("Configuration is not set up. Please run 'config set' command first to configure the CLI")
+	}
+
+	if extension.Host == "" {
+		return nil, fmt.Errorf("API server host is not configured. Please run 'config set' command first to configure the CLI")
+	}
```

Comment:
Would it make sense for these checks to be in a method on the config? Something like:
```go
clientConfig := c.Get()
if err := clientConfig.Validate(); err != nil {
    return nil, fmt.Errorf("invalid config. You may need to run the 'config set' command to configure the CLI: %w", err)
}

return client.NewRESTClient(clientConfig)
```